pkg/virtctl/vm/vm.go

```diff
@@ -395,10 +413,110 @@ func removeVolume(vmiName, volumeName, namespace string, virtClient kubecli.Kube
 	return nil
 }
 
+func calcMemoryDumpNeededSize(vmName, namespace string, virtClient kubecli.KubevirtClient) (*resource.Quantity, error) {
+	vmi, err := virtClient.VirtualMachineInstance(namespace).Get(vmName, &metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	domain := vmi.Spec.Domain
+	vmiMemoryReq := domain.Resources.Requests.Memory()
+	memOverhead := resource.MustParse(memoryDumpOverhead)
+	expectedPvcSize := resource.NewScaledQuantity(vmiMemoryReq.ScaledValue(resource.Kilo), resource.Kilo)
+	expectedPvcSize.Add(memOverhead)
+
+	return expectedPvcSize, nil
+}
+
+func generatePVC(size *resource.Quantity, claimName, namespace, storageClass, accessMode string) (*k8sv1.PersistentVolumeClaim, error) {
+	pvc := &k8sv1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      claimName,
+			Namespace: namespace,
+		},
+		Spec: k8sv1.PersistentVolumeClaimSpec{
+			Resources: k8sv1.ResourceRequirements{
+				Requests: k8sv1.ResourceList{
+					k8sv1.ResourceStorage: *size,
+				},
+			},
+		},
+	}
+
+	if storageClass != "" {
+		pvc.Spec.StorageClassName = &storageClass
+	}
+
+	if accessMode == string(k8sv1.ReadOnlyMany) {
+		return nil, fmt.Errorf("cannot dump memory to a readonly pvc, use either ReadWriteOnce or ReadWriteMany if supported")
+	}
+	if accessMode != "" {
+		pvc.Spec.AccessModes = []k8sv1.PersistentVolumeAccessMode{k8sv1.PersistentVolumeAccessMode(accessMode)}
+	} else {
+		pvc.Spec.AccessModes = []k8sv1.PersistentVolumeAccessMode{k8sv1.ReadWriteOnce}
+	}
+
+	return pvc, nil
+}
+
+func createPVCforMemoryDump(namespace, vmName, claimName string, virtClient kubecli.KubevirtClient) error {
+	_, err := virtClient.CoreV1().PersistentVolumeClaims(namespace).Get(context.Background(), claimName, metav1.GetOptions{})
+	if err == nil {
+		return fmt.Errorf("PVC %s/%s already exists, check if it should be created if not remove create flag\n", namespace, claimName)
+	}
+	if !k8serrors.IsNotFound(err) {
+		return err
+	}
+
+	neededSize, err := calcMemoryDumpNeededSize(vmName, namespace, virtClient)
+	if err != nil {
+		return err
+	}
+
+	pvc, err := generatePVC(neededSize, claimName, namespace, storageClass, accessMode)
+	if err != nil {
+		return err
+	}
+
+	_, err = virtClient.CoreV1().PersistentVolumeClaims(namespace).Create(context.Background(), pvc, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	err = wait.PollImmediate(time.Second*2, time.Second*60, func() (bool, error) {
```

Comment:
Not sure this is necessary, If create does not return an error, the PVC should exist.  And it's not like we're waiting for some condition on the PVC

Reply:
I wasn't sure if this is needed or not. If create doesnt return an error it means the PVC should exist or it should be created? is it possible that it will take some time until it is actually created?

Reply:
virtctl is not using any caching so a Get followed by Create will return the resource.  Sometimes in controllers we wait until the resource is cached

Reply:
ok sounds good